ir/src/validation/error.rs

```diff
@@ -229,7 +232,7 @@ impl SemanticError {
 
     pub(crate) fn invalid_list_folding(
         lf_value_type: &air_script_core::ListFoldingValueExpr,
-        symbol_type: &SymbolType,
+        symbol_type: &SymbolBinding,
```

Comment:
question: Is it intentional to keep the `symbol_type` name here and everywhere else?